### PR TITLE
Detect, and use, the type of image file; prevents issues when file renamed to an improper extension

### LINKED ISSUE #19: Crashes when image extension doesn't match image type
If you create a png file mistake.png , then rename it to mistake.jpg, add it to a product, then this causes problems.

The product page will not show the price/add to cart buttons.  Also, in my implementation at least, it messes up the product before due to the fetch next product image function crashing as well.  The product search results page is also similarly killed.

Unfortunately it doesn't leave any trace in the logs either.

### PR DESCRIPTION
**Pull request by brittainmark**
fixes #19 and [forum issue](https://www.zen-cart.com/showthread.php?222983-Image-Handler-5-(for-v1-5-5-v1-5-8)-Support-Thread&p=1392819#post1392819)
Use image info returned by getimagesize in function calculate_size to determine the type of image to use in the function load_imageGD

### CODE REVIEW

**Comment by lat9** [comment]
Wrong repository, the active repository is https://github.com/lat9/zen_Image-Handler.  Note also that I've indicated that I'm not a fan of this.

**Comment by lat9** [comment]
> Wrong repository, the active repository is https://github.com/lat9/zen_Image-Handler. Note also that I've indicated that I'm not a fan of this.

Sigh, I can't keep up with my _own_ development.  This _is_ the active repository.

**Comment by brittainmark** [comment]
Yes I noted that but this does  not require any additional calls to getimagesize just utilises the one already in existence. Or was your objection more fundmental?

File: includes/classes/bmz_image_handler.class.php [hunk 1]
```diff
@@ -505,7 +506,7 @@ public function calculate_size($pref_width, $pref_height = '')
                 $this->filename = DIR_WS_IMAGES . PRODUCTS_IMAGE_NO_IMAGE;
                 $image_info = getimagesize($this->filename);
             }
-            list($width, $height) = $image_info;
+            list($width, $height, $this->image_type) = $image_info;
             $this->ihLog("calculate_size: file " . $this->filename . " ($pref_width, $pref_height), getimagesize returned $width x $height.");
```
Comment (lat9) [inline includes/classes/bmz_image_handler.class.php:510]: Please include the determined image_type as part of the ihLog information.

File: includes/classes/bmz_image_handler.class.php [hunk 2]
```diff
@@ -927,25 +928,24 @@ protected function calculate_gravity($canvaswidth, $canvasheight, $overlaywidth,
      *
      * @return false|GdImage|mixed|resource
      */
-    protected function load_imageGD($src_name)
+    protected function load_imageGD($src_name,$retry = false)
```
Comment (lat9) [inline includes/classes/bmz_image_handler.class.php:931]: Where's the usage of this newly-added function parameter?  Either use it (and add a space after the `,`) or remove it.

**Review by lat9 — CHANGES_REQUESTED** [review]
Also, update the top-of-module comments for the class to indicate that it's part of 'Image Handler 5.3.2'.

**Comment by brittainmark** [comment]
replaced with #270

**Pull request closed without merging**